### DISCUSSION TITLE
Host on NAS, process on desktop, shared album

### DISCUSSION THREAD

**Discussion by drbubbles** [immich-app/immich#4019] [Q&A] [answered]
### The feature

**Feature detail**
Would it be straightforward to have a NAS and a PC  share a common immich gallery, so the PC can do the hard work populating the gallery and analyzing images, while the NAS hosts the gallery and can respond to search requests based on the results of the ML work that the PC did?

**Platform**
Server

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment, marked as answer]
This is definitely possible, but not necessarily straightforward depending on how experienced you are with self-hosting. Immich is built as a few separate containers, and those don't necessarily need to run on the same machine. The bulk of the hard work happens in microservices and machine-learning.

Running ML on a separate system is fairly easy and is documented at https://immich.app/docs/guides/machine-learning.

Running the microservices somewhere else is a little more involved, as you need to get it access to other components of the stack: postgres, redis, typesense, machine-learning, and of course the library folder. There are many ways to arrange that, so the details would depend on your preferences and how your systems are set up.

**Reply by AngelaDMerkel** [reply]
This is relatively easy to do with docker, but requires some preexisting knowledge. It is also best done using [docker swarm](https://docs.docker.com/engine/swarm/), but that only makes real sense in a server setting. If this is a desktop which will drop in and out of service, this doesn't make sense

**Comment by drbubbles** [comment]
I missed that link in the docs, thank you @bo0tzz, this looks like the best route for me. Yes @AngelaDMerkel  the desktop will drop in and out of service, while the NAS will be on all the time (or by wol) and accessible remotely via VPN while on travel for example. I'd never seen docker swarm before, amazing!
